src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveSubscription.java

```diff
@@ -182,8 +184,12 @@ static class State {
 		 */
 		Mono<Void> subscribe(ByteBuffer[] targets, Function<ByteBuffer[], Mono<Void>> subscribeFunction) {
 
-			return subscribeFunction.apply(targets).doOnSuccess((discard) -> this.targets.addAll(Arrays.asList(targets)))
-					.onErrorMap(exceptionTranslator);
+			return subscribeFunction.apply(targets).doOnSuccess((discard) -> {
+
+				for (ByteBuffer target : targets) {
```

Comment:
Maybe we could introduce a wrapper object that can do the wrapping of the `ByteBuffer` instead of adding to the `Set` directly.

Reply:
The entire buffer handling isn't ideal. I currently don't have a better idea, Buffers and byte arrays aren't great as map or set keys.